packages/binding.template/spec/nativeTemplateEngineBehaviors.ts

```diff
@@ -141,17 +141,25 @@ describe('Native template engine', function () {
         + "(Val: <span data-bind='text: $data'></span>, Invocations: <span data-bind='text: $root.invocationCount()'></span>, Parents: <span data-bind='text: $parents.length'></span>)"
         + '</div>'
         + '</div>'
-      const viewModel = {
-        invocations: 0, // Verifying # invocations to be sure we're not rendering anything multiple times and discarding the results
-        items: observableArray([
-          { children: observableArray(['A1', 'A2', 'A3']) },
-          { children: observableArray(['B1', 'B2']) }
-        ]),
-        invocationCount: undefined
+
+      class myViewModel {
+        invocations: number = 0 // Verifying # invocations to be sure we're not rendering anything multiple times and discarding the results
+        items: ObservableArray<any>
+
+        constructor() {
+          this.items = observableArray([
+            { children: observableArray(['A1', 'A2', 'A3']) },
+            { children: observableArray(['B1', 'B2']) }
+          ])
+        }
+
+        invocationCount = function () {
```

Comment:
the `this` inside `function` will be the `this` of the anonymous `function ()`, solved with a method or lambda e.g.

```ts
invocationCount () { ++this.invocations } 
// or
invocationCount = () => ++this.invocations
```

I would expect that invocations will be `0` no matter what here.  It's not clear from the code review whether invocations being > 0 is validated.

Reply:
fixed -> `invocationCount() {
          return ++this.invocations
        }`